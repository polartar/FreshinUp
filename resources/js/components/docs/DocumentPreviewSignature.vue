<template>
  <v-layout
    column
    mb-3
  >
    <v-layout>
      <v-flex
        v-for="item in signatureItems"
        :key="item.name"
        sm4
      >
        <document-preview-signature-item
          :item="item.name"
          :class="item.class"
        />
      </v-flex>
    </v-layout>
  </v-layout>
</template>

<script>
import DocumentPreviewSignatureItem from '~/components/docs/DocumentPreviewSignatureItem'

export default {
  components: {
    DocumentPreviewSignatureItem
  },
  props: {
    signeeName: {
      type: String,
      required: true
    }
  },
  data() {
    return {
      signatureItems: [
        {
          name: this.signeeName,
          class: 'mr-3'
        },
        {
          name: 'Printed Name',
          class: 'mx-3'
        },
        {
          name: 'Date',
          class: 'ml-3'
        }
      ]
    }
  },
}

</script>
