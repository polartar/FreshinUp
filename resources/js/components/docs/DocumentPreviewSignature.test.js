import * as Stories from './DocumentPreviewSignature.stories'
import Component from './DocumentPreviewSignature'
import { mount, shallowMount } from '@vue/test-utils'

describe('component/doc/DocumentPreviewSignature', () => {
  describe('Snapshots', () => {
    test('Default', async () => {
      const wrapper = mount(Stories.Default())
      await wrapper.vm.$nextTick()
      await wrapper.vm.$nextTick()
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('Props & Computed', () => {
    test('signeeName', async () => {
      const signeeName = 'moc signeeName'
      const wrapper = shallowMount(Component, {
        propsData: {
          signeeName
        }
      })
      expect(wrapper.vm.signeeName).toMatch(signeeName)
    })
  })
})
