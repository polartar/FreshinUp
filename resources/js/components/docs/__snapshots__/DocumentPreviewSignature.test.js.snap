// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component/doc/DocumentPreviewSignature Snapshots Default 1`] = `
<div
  class="container"
>
  <div
    class="layout column mb-3"
  >
    <div
      class="layout"
    >
      <div
        class="flex sm4"
      >
        <div
          class="mr-3"
        >
          <div
            class="py-3"
          />
           
          <hr
            class="v-divider theme--light"
          />
           
          <div
            class="body-1 mt-2"
          >
            
    mock owner name
  
          </div>
        </div>
      </div>
      <div
        class="flex sm4"
      >
        <div
          class="mx-3"
        >
          <div
            class="py-3"
          />
           
          <hr
            class="v-divider theme--light"
          />
           
          <div
            class="body-1 mt-2"
          >
            
    Printed Name
  
          </div>
        </div>
      </div>
      <div
        class="flex sm4"
      >
        <div
          class="ml-3"
        >
          <div
            class="py-3"
          />
           
          <hr
            class="v-divider theme--light"
          />
           
          <div
            class="body-1 mt-2"
          >
            
    Date
  
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
